### PR TITLE
feat: refactor JavaScript implementation and add C `ndarray` implementation for `blas/base/dnrm2`

### PR DESCRIPTION
**Pull request by aman-095**
Progresses #2039.

## Description

> What is the purpose of this pull request?

This pull request:

-   updates Js implementation based on the current updated `Lapack` implementation.
-   adds C ndarray API for `dnrm2`.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   progresses #2039  

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/base/dnrm2/src/dnrm2.f [hunk 1]
```diff
@@ -52,6 +52,20 @@
 !<
 double precision function dnrm2( N, dx, stride )
   implicit none
+  ! ..
+  ! Define a kind parameter for single precision:
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/dnrm2/src/dnrm2.f]:
```suggestion
  ! Define a kind parameter for double-precision:
```

File: lib/node_modules/@stdlib/blas/base/dnrm2/src/dnrm2.f [hunk 2]
```diff
@@ -52,6 +52,20 @@
...
+  integer, parameter :: wp = kind( 1.0d0 )
+  ! ..
+  ! Define constants:
+  real( wp ), parameter :: zero = 0.0_wp
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/dnrm2/src/dnrm2.f:60]: Is `real` the right function here? Based on https://gcc.gnu.org/onlinedocs/gfortran/REAL.html, I am not sure.

**Review by kgryte — COMMENTED** [review]
Apart from my comment regarding `real(kind)`, this PR is looking good.

File: lib/node_modules/@stdlib/blas/base/dnrm2/src/dnrm2.f (hunk 2 above)
Comment (kgryte) [reply lib/node_modules/@stdlib/blas/base/dnrm2/src/dnrm2.f:60]: What does LAPACK use here?

Comment (aman-095) [reply lib/node_modules/@stdlib/blas/base/dnrm2/src/dnrm2.f:60]: Based on the `lapack` implementation this is correct: https://www.netlib.org/lapack/explore-html/d1/d2a/group__nrm2_gab5393665c8f0e7d5de9bd1dd2ff0d9d0.html#gab5393665c8f0e7d5de9bd1dd2ff0d9d0

Comment (aman-095) [reply lib/node_modules/@stdlib/blas/base/dnrm2/src/dnrm2.f:60]: Also, we have used a similar approach for `dznrm2` as well: https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/blas/base/dznrm2/src/dznrm2.f#L58C1-L59C1

Comment (kgryte) [reply lib/node_modules/@stdlib/blas/base/dnrm2/src/dnrm2.f:60]: Thanks for confirming.

**Review by kgryte — APPROVED** [review]

**Pull request merged**